include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -438,4 +467,45 @@ def TT_AssertOp : TT_Op<"assert", [MemoryEffects<[MemWrite]>]> {
   let assemblyFormat = "$condition `,` $message `,` $file `,` $func `,` $line attr-dict `:` type($condition)";
 }
 
+//
+// Make a Tile Pointer
+//
+def TT_MakeTilePtrOp : TT_Op<"make_tile_ptr",
+                             [Pure,
+                              SameVariadicOperandSize,
+                              TypesMatchWith<"infer pointer type from the result type",
+                                             "result", "base",
+                                             "getPointerType(getElementTypeOfTilePointerType($_self))">]> {
+  let summary = "Make a tile pointer type with meta information of the parent tensor and the tile";
+
+  let description = [{
+      `tt.make_tile_ptr` takes both meta information of the parent tensor and the tile and returns a pointer to a tile,
+      e.g. returns a type of `tt.ptr<tensor<8x8xf16>>`.
+  }];
+
+  let arguments = (ins
+    TT_Ptr:$base,
+    Variadic<I64>:$shape,
+    Variadic<I64>:$strides,
+    Variadic<I32>:$offsets,
```

Comment:
It's actually wierd that shape/stride is I64 while offset is I32. (it's useless for strides/offsets to be i64 if offset can only be i32). The issues that causing this has not been fully fixed for now. I think we could regard this as some sort of TODO in the internal op definition, and we don't expose the unreasonable restriction (offsets is I32) to the users. Do you agree ? @ptillet

Reply:
Or maybe     
```
Variadic<I64orI32>:$shape,
Variadic<I64orI32>:$strides,
Variadic<I64orI32>:$offsets,
```
makes more sense, in which they should always have identical interger widths.

Reply:
I think there's some very confusing terminology here. `shape` makes it seem at a glance like this is the result triton tensor shape, but it's actually the size of the parent tensor that's being tiled.

How about:
```
Variadic<I64orI32>:$parentShape,
Variadic<I64orI32>:$parentStrides,
Variadic<I64orI32>:$parentOffsets,
```
?

Reply:
I also think `parentShape/...` is better. But the problem we want to discuss is the type I64/I32. It's actually weird that shape/stride is I64 while offset is I32 (for some `sm_90` reasons).

Reply:
Ah, I didn't know `offset had to be I32 to accomodate the hardware. But I think there are still some cases where it would make sense for the offset to be i32 (like 0) while the stride is i64 (for example some 2x2**33 parent tensor)?

Reply:
Yeah, that makes sense. I will just make the current types and add a note, not a big deal.